pymeasure/adapters/prologix.py

```diff
@@ -168,13 +175,20 @@ def gpib(self, address, rw_delay=None):
         rw_delay = rw_delay or self.rw_delay
         return PrologixAdapter(self.connection, address, rw_delay=rw_delay)
 
-    def wait_for_srq(self, timeout=25, delay=0.1):
+    def _check_for_srq(self, rw_delay):
+        # it was int(self.ask("++srq"))
+        self.write("++srq")
+        time.sleep(rw_delay or self.rw_delay)
+        return int(self.read())
+
+    def wait_for_srq(self, timeout=25, delay=0.1, rw_delay=0):
         """ Blocks until a SRQ, and leaves the bit high
 
-        :param timeout: Timeout duration in seconds
-        :param delay: Time delay between checking SRQ in seconds
+        :param timeout: Timeout duration in seconds.
+        :param delay: Time delay between checking SRQ in seconds.
+        :param rw_delay: Time delay between writing and reading in seconds.
         """
-        while int(self.ask("++srq")) != 1:  # TODO: Include timeout!
+        while self._check_for_srq(rw_delay) != 1:  # TODO: Include timeout!
```

Comment:
While we're here, can we deal with that TODO?

Reply:
Do you know, what the idea of the implementation was?
Breaking the while loop after timeout time?
What to do in that case? Raise a connection Error?

Reply:
no idea. @cjermain @msmttchr ?